aea/protocols/generator/generator.py

```diff
@@ -2405,50 +2012,61 @@ def _generate_file(self, file_name: str, file_content: str) -> None:
         with open(pathname, "w") as file:
             file.write(file_content)
 
-    def generate(self) -> None:
+    def generate(self, protobuf_only: bool = False) -> None:
         """
-        Create the protocol package with Message, Serialization, __init__, protocol.yaml files.
+        Run the generator. If in "full" mode (protobuf_only is False), it:
+        a) validates the protocol specification.
```

Comment:
Validation seems to be happening elsewhere, I cannot see it below.

Reply:
Validation is a prerequisite for extraction so `extract()` already calls `validate()`. 

The docstring gives an overall explanation of major things that happen in the two modes.